### ISSUE TITLE
`gem install fluentd:0.14.2` installation broken because of recent serverengine:2.0.0 release

### ISSUE THREAD

**Issue by icio** [fluent/fluentd#1195]
fluentd 0.14.2 specifies the dependency [`gem.add_runtime_dependency("serverengine", [">= 1.6.4"])`](https://github.com/fluent/fluentd/blob/1ef51cbfcc51e24d923173c27b7adf7724c6979a/fluentd.gemspec#L26) which resolves to the recently-released [serverengine:2.0.0](https://rubygems.org/gems/serverengine/versions/2.0.0), but fluentd 0.14.2 is not compatible with serverengine 2.0.0.

The installation succeeds (note: serverengine-2.0.0)

```bash
# gem install fluentd:0.14.2 fluent-plugin-systemd fluent-plugin-elasticsearch | grep installed
Successfully installed msgpack-1.0.0
Successfully installed yajl-ruby-1.2.1
Successfully installed cool.io-1.4.4
Successfully installed sigdump-0.2.4
Successfully installed serverengine-2.0.0
Successfully installed http_parser.rb-0.6.0
Successfully installed thread_safe-0.3.5
Successfully installed tzinfo-1.2.2
Successfully installed tzinfo-data-1.2016.6
Successfully installed strptime-0.1.8
Successfully installed fluentd-0.14.2
Successfully installed ffi-1.9.14
Successfully installed systemd-journal-1.2.2
Successfully installed fluent-plugin-systemd-0.0.4
Successfully installed excon-0.52.0
Successfully installed multi_json-1.12.1
Successfully installed multipart-post-2.0.0
Successfully installed faraday-0.9.2
Successfully installed elasticsearch-transport-1.0.18
Successfully installed elasticsearch-api-1.0.18
Successfully installed elasticsearch-1.0.18
Successfully installed fluent-plugin-elasticsearch-1.6.0
22 gems installed
```

But the installation is broken:

```bash
# fluentd --version
fluentd 0.14.2

# fluentd -s /tmp/fluent/
Installed /tmp/fluent/fluent.conf.

# fluentd -c /tmp/fluent/fluent.conf
2016-08-30 17:18:27 +0000 [info]: reading config file path="/tmp/fluent/fluent.conf"
2016-08-30 17:18:27 +0000 [info]: starting fluentd-0.14.2
2016-08-30 17:18:27 +0000 [info]: spawn command to main: /usr/bin/ruby2.3 -Eascii-8bit:ascii-8bit /usr/local/bin/fluentd -c /tmp/fluent/fluent.conf --under-supervisor
2016-08-30 17:18:28 +0000 [info]: reading config file path="/tmp/fluent/fluent.conf"
2016-08-30 17:18:28 +0000 [info]: starting fluentd-0.14.2 without supervision
2016-08-30 17:18:28 +0000 [error]: unexpected error error="undefined method `change_privilege' for ServerEngine::Daemon:Class"
  2016-08-30 17:18:28 +0000 [error]: /var/lib/gems/2.3.0/gems/fluentd-0.14.2/lib/fluent/supervisor.rb:601:in `change_privilege'
  2016-08-30 17:18:28 +0000 [error]: /var/lib/gems/2.3.0/gems/fluentd-0.14.2/lib/fluent/supervisor.rb:406:in `block in run_worker'
  2016-08-30 17:18:28 +0000 [error]: /var/lib/gems/2.3.0/gems/fluentd-0.14.2/lib/fluent/supervisor.rb:553:in `main_process'
  2016-08-30 17:18:28 +0000 [error]: /var/lib/gems/2.3.0/gems/fluentd-0.14.2/lib/fluent/supervisor.rb:404:in `run_worker'
  2016-08-30 17:18:28 +0000 [error]: /var/lib/gems/2.3.0/gems/fluentd-0.14.2/lib/fluent/command/fluentd.rb:271:in `<top (required)>'
  2016-08-30 17:18:28 +0000 [error]: /usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
  2016-08-30 17:18:28 +0000 [error]: /usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
  2016-08-30 17:18:28 +0000 [error]: /var/lib/gems/2.3.0/gems/fluentd-0.14.2/bin/fluentd:5:in `<top (required)>'
  2016-08-30 17:18:28 +0000 [error]: /usr/local/bin/fluentd:23:in `load'
  2016-08-30 17:18:28 +0000 [error]: /usr/local/bin/fluentd:23:in `<main>'
2016-08-30 17:18:28 +0000 [info]: Worker 0 finished unexpectedly with status 1
...
```

**Comment by warmfusion** [comment]
This appears to have been fixed by the following as yet unreleased change : 

https://github.com/fluent/fluentd/blob/37fb0a44eea7af6b8bf81a6a79c9b1c2d8b36fa5/lib/fluent/supervisor.rb

**Comment by icio** [comment]
Replacing serverengine 2.0.0 with 1.6.4 resolves the issue:

```
# gem uninstall serverengine
...
Successfully uninstalled serverengine-2.0.0

# gem install serverengine -v '<2.0.0'
Successfully installed serverengine-1.6.4
...

# fluentd -c /tmp/fluent/fluent.conf
2016-08-30 17:22:47 +0000 [info]: reading config file path="/tmp/fluent/fluent.conf"
2016-08-30 17:22:47 +0000 [info]: starting fluentd-0.14.2
2016-08-30 17:22:47 +0000 [info]: spawn command to main: /usr/bin/ruby2.3 -Eascii-8bit:ascii-8bit /usr/local/bin/fluentd -c /tmp/fluent/fluent.conf --under-supervisor
2016-08-30 17:22:48 +0000 [info]: reading config file path="/tmp/fluent/fluent.conf"
2016-08-30 17:22:48 +0000 [info]: starting fluentd-0.14.2 without supervision
2016-08-30 17:22:48 +0000 [info]: gem 'fluent-plugin-elasticsearch' version '1.6.0'
2016-08-30 17:22:48 +0000 [info]: gem 'fluent-plugin-systemd' version '0.0.4'
2016-08-30 17:22:48 +0000 [info]: gem 'fluentd' version '0.14.2'
2016-08-30 17:22:48 +0000 [info]: adding match pattern="debug.**" type="stdout"
...
```

**Comment by icio** [comment]
I suggest a quick deploy of #1196 tagged under 0.14.3 to resolve this issue, despite the changes to support serverengine 2.0.0, as noted by @warmfusion above.

**Comment by repeatedly** [comment]
Fixed by v0.14.3. Thanks for the report.

**Comment by icio** [comment]
🎉  Nice one
